### PR TITLE
many: move to go modules 

### PR DESCRIPTION
**Pull request by mvo5**
This is a variant of https://github.com/snapcore/snapd/pull/10559 which a more gradual approach. It keeps the vendor.json and vendor dir for a little bit to see if things can be move gradually.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+require (
+	github.com/boltdb/bolt v1.3.1 // indirect
+	github.com/canonical/go-sp800.90a-drbg v0.0.0-20210314144037-6eeb1040d6c3 // indirect
+	github.com/canonical/go-tpm2 v0.0.0-20210314160024-32171bd353b1
+	github.com/canonical/tcglog-parser v0.0.0-20200908165021-12a3a7bcf5a1 // indirect
+	github.com/coreos/go-systemd v0.0.0-20180511133405-39ca1b05acc7
+	github.com/godbus/dbus v0.0.0-20190726142602-4481cbc300e2
+	github.com/gorilla/mux v1.7.4-0.20190701202633-d83b6ffe499a
+	github.com/gvalkov/golang-evdev v0.0.0-20191114124502-287e62b94bcb
```
Comment (anonymouse64) [inline go.mod:13]: I'm surprised if this works, it didn't work for me when I tried building with upstream because the tests in that package are broken

Comment (mvo5) [reply go.mod:13]: it's not working yet :/

Comment (mvo5) [reply go.mod:13]: It seems it is working afterall, no idea why it works here but did not for you.

**Review by anonymouse64 — COMMENTED** [review]
lgtm thus far

**Comment by bboozzoo** [comment]
I'll look into why opensuse is failing.

File: packaging/ubuntu-14.04/rules [hunk 2]
```diff
@@ -212,10 +212,19 @@ override_dh_install:
 override_dh_auto_install: snap.8
 	dh_auto_install -O--buildsystem=golang
 
+override_dh_golang:
+	# XXX: dh_golang on older versions of ubuntu is not capable to deal
+	# deal with newer go and go.mod. So just make this a no-op or things
```
Comment (mardy) [inline packaging/ubuntu-14.04/rules:217]: there's a duplicate "deal" here, but let's pretend we didn't see it :-)

**Review by mardy — APPROVED** [review]
I understand 5% of this, but that 5% is very correct :-)

**Review by anonymouse64 — APPROVED** [review]
lgtm, I think this is good, thank you for working on it more and getting it this far!

**Pull request merged**